pkg/tools/gen/gendoc.go

```diff
@@ -211,23 +232,29 @@ func (pkg *KclPackage) getPackageIndexContent(level int, indentation string) str
 }
 
 func (tpe *KclOpenAPIType) getSchemaIndexContent(level int, indentation string) string {
-	return fmt.Sprintf(`%s- [%s](#%s)
-`, strings.Repeat(indentation, level), tpe.KclExtensions.XKclModelType.Type, strings.ToLower(tpe.KclExtensions.XKclModelType.Type))
+    var referencedBy strings.Builder
```

Comment:
Why add the `referenced by` info on the TOC?

Reply:
Yes, you are right. We want the TOC to be easy to read and understand. Initially, I thought adding `referenced by` info to the TOC would be helpful because it could provide additional context about where certain schema types are used.

Reply:
A single section named `referenced by` maybe useful.